src/main/java/com/onarandombox/MultiverseCore/listeners/MVPlayerListener.java

```diff
@@ -105,26 +117,32 @@ private Location getMostAccurateRespawnLocation(World w) {
      */
     @EventHandler
     public void playerJoin(PlayerJoinEvent event) {
-        Player p = event.getPlayer();
-        if (!p.hasPlayedBefore()) {
+        Player player = event.getPlayer();
+        if (!player.hasPlayedBefore()) {
             Logging.finer("Player joined for the FIRST time!");
-            if (plugin.getMVConfig().getFirstSpawnOverride()) {
-                Logging.fine("Moving NEW player to(firstspawnoverride): "
-                        + worldManager.getFirstSpawnWorld().getSpawnLocation());
-                this.sendPlayerToDefaultWorld(p);
+            if (!plugin.getMVConfig().getFirstSpawnOverride()) {
+                return;
             }
+            Logging.fine("Moving NEW player to(firstspawnoverride): " + worldManager.getFirstSpawnWorld().getSpawnLocation());
+            this.sendPlayerToDefaultWorld(player);
+            return;
+        }
+
+        Logging.finer("Player joined AGAIN!");
+        MVWorld world = worldManager.getMVWorld(player.getWorld());
+        if (world == null) {
+            Logging.fine("Player is in a world that is not managed by Multiverse. Ignoring.");
```

Comment:
This looks like new behavior. Is that right?

Reply:
yea, but it makes more sense bcu access permission isnt checked when a player teleports to a non-mv world (even before this PR). So I don't see why MV should check for access permission of a non-mv world on player join.